wgsl/index.bs

```diff
@@ -5798,6 +5798,67 @@ reduce a shader's memory bandwidth demand.
         See [[#floating-point-conversion]] for edge case behaviour.
 </table>
 
+## Synchronization built-in functions ## {#sync-builtin-functions}
+
+[SHORTNAME] provides a single synchronization function:
+
+```rust
+workgroupBarrier<storage_class_list>() -> void
+```
+
+Execute a control barrier with Acquire/Release memory ordering. That is, all
+affected memory, atomic and barrier operations are ordered in
+[[#program-order]] relative to the barrier. Additionally, the affected memory
+and atomic operations program ordered before the workgroupBarrier must be
+visible to all other threads in the workgroup before any affected memory or
+atomic operation program ordered after the workgroupBarrier is executed by a
+member of the workgroup.
+
+A memory, atomic or workgroupBarrier operation is an afftected operation if it
```

Comment:
Do we have a description of what a "memory" operation is? I.e. is loading from a storage texture a memory operation?

Reply:
If read_write textures are eventually added, then barriers would need to be updated to allow the handle storage class, and yes, would be considered memory operations. To me, the right place to describe memory operations would be in the memory model section which is still TODO; however, we could say that memory operations are a read or write to a variable (or through a pointer) in an affected storage class if that is preferable.

Reply:
Hold on. So today without read/write textures, if I do two different `textureStore` operations, there are no barriers I could use to ensure that one is done after the other?

Reply:
Correct. Metal doesn't add image barriers widely until Metal 2.0 so I didn't want to force that requirement on WGSL. I think that without read_write textures, the problem you describe isn't that relevant. It would mean you're overwriting the same pixel from multiple invocations without any interim reads. I don't think that's the common case.

I'd be happy to add texture coverage if we knew only Metal 2.0 or later was being targeted with WGSL.